PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/webapp/action/CreateWebAppAction.java

```diff
@@ -81,52 +81,42 @@ public void actionPerformed(NodeActionEvent e) {
         this.openDialog(project, null);
     }
 
-    private void openDialog(final Project project, @Nullable WebAppConfig config) {
-        final WebAppCreationDialog dialog = new WebAppCreationDialog(project);
-        dialog.setOkActionListener((data) -> {
-            this.createWebApp(data, project, dialog);
+    private void openDialog(final Project project, @Nullable final WebAppConfig data) {
+        final WebAppCreationDialog dialog = this.createDialog(project, data);
+        dialog.setOkActionListener((config) -> {
+            dialog.close();
+            this.createWebApp(config)
+                .subscribe(webapp -> {
+                    final Path artifact = config.getApplication();
+                    if (Objects.nonNull(artifact) && artifact.toFile().exists()) {
+                        AzureTaskManager.getInstance().runLater("deploy", () -> deploy(webapp, artifact, project));
+                    }
+                }, (error) -> {
+                    final AzureExceptionAction action = AzureExceptionAction.simple(
+                        String.format("Reopen dialog \"%s\"", dialog.getTitle()),
+                        t -> AzureTaskManager.getInstance().runLater("open dialog", () -> this.openDialog(project, config)));
```

Comment:
Looks unintelligible. try to consider if there is any approach to refactor it to more readable?

Reply:
I am considering migrate AzureExceptionAction to AzureTask. 
so in the future, it would something like:
```
Messager.error(exception, new AzureTask("reopen dialog", ()->this.openDialog(project, config));
```